mirrorlist/mirrorlist_server.py

```diff
@@ -161,7 +162,7 @@ def metalink(cache, directory, file, hosts_and_urls):
     preference = 100
     try:
         fdc = file_details_cache[directory]
-        detailslist = fdc[file]
+        detailslist = sorted(fdc[file], key=itemgetter('timestamp'), reverse=True)
```

Comment:
While this works, I wonder if we could do the ordering sooner, maybe directly in the pickle so that there is no speed impact on the mirrorlist. What do you think?